src/ui/components/nav/components/UserMenu.tsx

```diff
@@ -0,0 +1,95 @@
+"use client";
+
+import { Fragment } from "react";
+import Link from "next/link";
+import Image from "next/image";
+import clsx from "clsx";
+import { useQuery } from "urql";
+import { UserIcon } from "lucide-react";
+import { Menu, Transition } from "@headlessui/react";
+import { useSaleorAuthContext } from "@saleor/auth-sdk/react";
+import { CurrentUserDocument, type CurrentUserQuery } from "@/gql/graphql";
+
+export function UserMenu() {
+	const { signOut } = useSaleorAuthContext();
+	const [{ data }] = useQuery<CurrentUserQuery>({
+		query: CurrentUserDocument.toString(),
+	});
+
+	if (data?.me) {
+		return (
+			<Menu as="div" className="relative ml-3">
+				<Menu.Button className="relative flex rounded-full bg-neutral-200 text-sm focus:outline-none focus:ring-2 focus:ring-white focus:ring-offset-2 focus:ring-offset-neutral-800">
+					<span className="sr-only">Open user menu</span>
+					{data.me.avatar ? (
+						<Image
+							className="h-8 w-8 rounded-full border"
+							aria-hidden="true"
+							src={data.me.avatar.url}
+							width={24}
+							height={24}
+							alt=""
+						/>
+					) : (
+						<span
+							className="flex h-8 w-8 items-center justify-center rounded-full border bg-white text-center text-xs font-bold uppercase"
+							aria-hidden="true"
+						>
+							{data.me.email.slice(0, 2)}
```

Comment:
We typically use initials for this. This could be a separate component because the logic quickly becomes complicated